_posts/2021-06-06-spring-camel.md

```diff
@@ -0,0 +1,422 @@
+    return camelContext.createConsumerTemplate();
+  }
+
+}
+
+```
+
+Here we have defined a REST endpoint in our `resource` class with a `GET` method for fetching products by category. We are invoking our Camel route inside the method by using the `producerTemplate` which we configured in our Spring configuration. 
+
+In our Spring configuration we have defined the `producerTemplate` and `consumerTemplate` by calling corresponding methods on the `CamelContext` which is available in the `ApplicationContext`.
+
+## Defining a Route with Splitter-Aggregator Enterprise Integration Pattern
+
+Let us now look at a route where we will use a Enterprise Integration Pattern.
+
+Camel provides implementations for many of the [Enterprise Integration Patterns](https://www.enterpriseintegrationpatterns.com/patterns/messaging/toc.html) from the book by Gregor Hohpe and Bobby Woolf. Here is a snippet of the list of those patterns :
+
+![EIP Snippet](/assets/img/posts/camel-spring/eip-snippet.png)
+
+
+### Selecting the Enterprise Integration Pattern (EIP)
+Before trying to build our own integration logic, we should look for the integration pattern most appropriate for fulfilling our use case. 
+
+Let us see an example of defining a route with the Splitter and Aggregate integration patterns.  Here we will consider a hypothetical scenario of building a REST API for an E-Commerce application for processing an order placed by a customer. We will expect our order processing API to perform the following steps:
+
+1. Fetch the list of items from the shopping cart 
+2. Fetch price of each order line item in the cart 
+3. Calculate the sum of prices of all order line items to generate the order invoice.
+
+After finishing step 1, we want to fetch the price of each order line item in parallel in step 2, since they are not dependent on each other. There are multiple ways of doing this kind of processing. 
+
+However, since design patterns are accepted solutions to recurring problems within a given context, we will search for a pattern closely resembling our problem from our list of Enterprise Integration Patterns. After looking through the list, we find that the [Splitter](https://www.enterpriseintegrationpatterns.com/patterns/messaging/Sequencer.html) and [Aggregator](https://www.enterpriseintegrationpatterns.com/patterns/messaging/Aggregator.html) patterns are best suited to do this processing.
+
+### Applying the Enterprise Integration Pattern (EIP)
+
+Next, we will refer to the Apache Camel's documentation for the chosen pattern's usage:
+
+![Splitter-Aggregator](/assets/img/posts/camel-spring/splitter-aggregator.png)
+
+We can split a single message into multiple fragments with the [Splitter](https://camel.apache.org/components/3.4.x/eips/split-eip.html) and process them individually. After that, we can use the [Aggregator](https://camel.apache.org/components/latest/eips/aggregate-eip.html) to combine those individual fragments into a single message. 
```

Comment:
Same here (also I'm not sure about copyright with the screenshot). This screenshot becomes obsolete when you describe the Splitter and Aggregator patterns in your own words in the intro section of the EIP chapter.